### ISSUE TITLE
Freerdp sound (audio) choppy on low bandwidth connection

### ISSUE THREAD

**Issue by ahoppin** [FreeRDP/FreeRDP-old#28]
Sorry if this report is incorrectly formatted, I'm a relative newbie.

Version: 0.7 from tarball source
Hardware: Asus A7n8x, Athlon 2100, 512mb memory; further limited test with Lenovo T43 (specs unknown)
OS: Wary Puppy 5.1.1
Connection: T1 lan and 1.5mb/360k ADSL
Command: xfreerdp -a 16 -z --plugin rdpsnd [win_7_machine_ip_addr]

When connected to Windows 7 server machine via ~15mb/s lan, sound is delivered to client machine cleanly.

When connected to same server via vpn (gpptp) via 1.5mp/s ADSL, sound is choppy.  Sound halts approximately once per second, then resumes, then halts again.  Each sound interruption accompanied by following errmsg on stdout:  

"wave_out_play: underrun occurred
ALSA lib pcm.c:7245:(snd_pcm_recover) underrun occured"

Tried -x m, no improvement.  -z flag or not, no discernible change.

Audio quality on slow ADSL link (when it's audible) suggests relatively high audio compression bit rate.

Thanks.  Please advise if I should provide further information.

**Comment by llyzs** [comment]
Since you are using version 0.7, it was old stable and the issue should have been already fixed in current git master. Would you like to verify it?

**Comment by ahoppin** [comment]
My error, apologies.  This is 0.8.2.

**Comment by llyzs** [comment]
yes but this is also a known issue in 0.8.2 which should have been fixed in git master some time ago.

**Comment by ahoppin** [comment]
Sorry to be an obvious noob, but I can't confirm this issue does not exist in the git master, as thus far I have not been able to successfully make the clone git master into an executable.  I'm making under Puppy 5.2.5, and it appears that the compiler is not correctly passing FreeRDP's option flags to the linker.  All goes smoothly until make tries to process ~/FreeRDP/channels/rdpdr/printer and then I get fatal errors from make.

   CCLD   printer.la
/usr/bin/ld: cannot find -lgssapi_krt5
collect2: ld returned 1 exit status
make [2]: **\* [printer.la] Error 1
make [2]: Leaving directory `/root/FreeRDP/channels/rdpdr/printer'
make [1]: *** [all-recursive] Error 1
make [1]: Leaving directory`/root/FreeRDP'

Solved that by adding a space between -l and gssapi_krt5 and other args in $CUPS_LIBS) in ~/FreeRDP/Makefile

However, I can't find the reference for this further make error:

   CCLD   printer.la
/usr/bin/ld: cannot find -l-Wl,-soname

I grepped every file in the tree for the string 'Wl,-soname' (find . -depth -exec grep 'Wl,-soname' {} \;) and can only find it in ~/FreeRDP/configure.  It doesn't seem to be in any of the Makefiles, unless I missed it.  I haven't been able to fix this.  

Probably PEBKAC, old System V and SCO Xenix hacker here, or should this be a separate issue?  Apologies again for my ignorance.  Perhaps I have no business being here and should just wait for a new release.
